test_gender_selection.py

```diff
@@ -1,33 +1,58 @@
+import time
+import os
 import pytest
-from selenium.webdriver.support.ui import WebDriverWait
-from selenium.webdriver.support import expected_conditions as EC
+from PIL import Image, ImageChops
 from selenium import webdriver
 
 
+def is_images_equal(img1, img2):
+    image_1 = Image.open(img1).convert('RGB')
+    image_2 = Image.open(img2).convert('RGB')
+    result = ImageChops.difference(image_1, image_2).getbbox()
+    if result is None:
+        return True
+    return False
+
 @pytest.fixture
 def driver():
     # Browser settings
     chrome_options = webdriver.ChromeOptions()
-    chrome_options.add_argument('--auto-open-devtools-for-tabs')
 
     with webdriver.Chrome(options=chrome_options) as driver:
         driver.maximize_window()
-        driver.implicitly_wait(3)
+        driver.implicitly_wait(10)
 
         yield driver
 
 
 # Test run
 class TestClass:
+
     def test_gender_selection_female(self, driver):
         driver.get("https://webglsamples.org/collectibles/index.html")
-        elem = driver.find_element('id', 'dollbaseFemale')
-        elem.click()
-        WebDriverWait(driver, 5).until(EC.presence_of_element_located(('id', 'canvas')))
-
+        driver.find_element('id', 'dollbaseFemale').click()
+        test_path = 'test_gender_selection_female_test.png'
+        driver.save_screenshot(test_path)
+        for x in range(10):
+            time.sleep(0.1)
+            path = 'test_gender_selection_female.png'
+            driver.save_screenshot(path)
+            if is_images_equal(test_path, path):
+                break
+            driver.save_screenshot(test_path)
+        os.remove(test_path)
 
     def test_gender_selection_male(self, driver):
         driver.get("https://webglsamples.org/collectibles/index.html")
-        elem = driver.find_element('id', 'dollbaseMale')
-        elem.click()
-        WebDriverWait(driver, 5).until(EC.presence_of_element_located(('id', 'canvas')))
\ No newline at end of file
+        driver.find_element('id', 'dollbaseMale').click()
+        test_path = 'test_gender_selection_male_test.png'
+        driver.save_screenshot(test_path)
+        for x in range(10):
+            time.sleep(0.1)
+            path = 'test_gender_selection_male.png'
+            driver.save_screenshot(path)
+            if is_images_equal(test_path, path):
+                break
+            driver.save_screenshot(test_path)
+        os.remove(test_path)
```

Comment:
And what about second image? Why don't you delete it?

Reply:
The TK test should end with a screenshot. In the implementation, it makes no sense to take this screenshot and make it again. Will be corrected in task 12.